kubetest/kubernetes.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os/exec"
+	"time"
+)
+
+// kubectlGetNodes lists nodes by executing kubectl get nodes, parsing the output into a nodeList object
+func kubectlGetNodes() (*nodeList, error) {
+	o, err := output(exec.CommandContext(kubetestContext, "kubectl", "get", "nodes", "-ojson"))
```

Comment:
If we do this then can we also refactor output() and friends to use the context? Right now it seems like we have multiple ways of accomplishing the same thing (both cancel the context and send a kill message

Reply:
Absolutely.  And I'll happily do that, though we'll have to decide whether we want to just call exec.CommandContext and accept the default cancellation, or whether we want the existing strategy where SIGINT escalates to SIGTERM.  (Or whether we can partially substitute things).